include/directx/d3dx12.h

```diff
@@ -4037,6 +4038,432 @@ class CD3DX12_NODE_MASK_SUBOBJECT
     D3D12_NODE_MASK m_Desc;
 };
 
+//------------------------------------------------------------------------------------------------
+// Peter: Implementation for the new CheckFeatureSupport API
+
+// Macro to set up a getter function for each entry in feature support data
+// The getter function will have the same name as the feature option name
+#define FEATURE_SUPPORT_GET(RETTYPE,FEATURE,OPTION) \
+RETTYPE OPTION() const \
+{ \
+    return FEATURE.OPTION; \
+}
+
+// Macro to set up a getter function for each entry in feature support data
+// Also specifies the name for the function which can be different from the feature name
+#define FEATURE_SUPPORT_GET_NAME(RETTYPE,FEATURE,OPTION,NAME) \
+RETTYPE NAME() const \
+{\
+    return FEATURE.OPTION; \
+}
+
+// Macro to set up a getter function for feature data indexed by the graphics node ID
+// The default parameter is 0, or the first availabe graphics device node
+#define FEATURE_SUPPORT_GET_NODE_INDEXED(RETTYPE,FEATURE,OPTION) \
+RETTYPE OPTION(UINT NodeIndex=0) const \
+{\
+    return FEATURE[NodeIndex].OPTION; \
+}
+
+// Macro to initialize a member feature data struct corresponding to the specified feature
+// Will stop the initialization process and return an error code upon failure
+#define INITIALIZE_MEMBER_DATA_CHECKED(FEATURE, MEMBER) \
+if (FAILED(m_hStatus = m_pDevice->CheckFeatureSupport(FEATURE, &MEMBER, sizeof(MEMBER)))) { \
+    return m_hStatus; \
+}
+
+// Macro to initialize a member feature data and report on failure
+// Does not stop the initialization process by itself
+#define INITIALIZE_FAILED(FEATURE, MEMBER) FAILED(m_hStatus = m_pDevice->CheckFeatureSupport(FEATURE, &MEMBER, sizeof(MEMBER)))
+
+class CD3DX12FeatureSupport
+{
+public:
+    CD3DX12FeatureSupport()
+    : m_pDevice(nullptr)
+    , m_hStatus(E_NOTIMPL)
+    {}
+
+    CD3DX12FeatureSupport(ID3D12Device* pDevice)
+    {
+        Init(pDevice);
+    }
+
+    HRESULT Init(ID3D12Device* pDevice)
+    {
+        if (!pDevice) {
+            m_hStatus = E_NOTIMPL;
+            return m_hStatus;
+        }
+
+        m_pDevice = pDevice;
+
+        
+        // Initialize static feature support data structures
+        if (INITIALIZE_FAILED(D3D12_FEATURE_D3D12_OPTIONS, m_dOptions)) {
+            m_dOptions.DoublePrecisionFloatShaderOps = false;
+            m_dOptions.OutputMergerLogicOp = false;
+            m_dOptions.MinPrecisionSupport = D3D12_SHADER_MIN_PRECISION_SUPPORT_NONE;
+            m_dOptions.TiledResourcesTier = D3D12_TILED_RESOURCES_TIER_NOT_SUPPORTED;
+            m_dOptions.ResourceBindingTier = (D3D12_RESOURCE_BINDING_TIER)0;

```

Comment:
Some enums have an entry that stands for NOT_SUPPORTED and it's usually 0. Others don't, but I still set them to 0 if the entry is not occupied.